benchmark/scenarios/test/test_write_request_load.py

```diff
@@ -0,0 +1,507 @@
+        """
+        WriteRequestLoadScenario raises RequestRateNotReached if the
+        target rate cannot be established within a given timeframe.
+        """
+        c = Clock()
+        cluster = self.make_cluster(
+            self.get_dropping_flocker_client_instance())
+        s = WriteRequestLoadScenario(c, cluster)
+        cluster.get_control_service(c).drop_requests = True
+        d = s.start()
+
+        # Continue the clock for one second longer than the timeout
+        # value to allow the timeout to be triggered.
+        c.advance(s.timeout + 1)
+
+        failure = self.failureResultOf(d)
+        self.assertIsInstance(failure.value, WRequestRateNotReached)
+
+    def test_scenario_throws_exception_if_overloaded(self):
+        """
+        `WriteRequestLoadScenario` raises `RequestOverload` if the
+        difference between sent requests and received requests exceeds
+        the tolerated difference once we start monitoring the scenario.
+
+        Note that, right now, the only way to make it fail is to generate
+        this difference before we start monitoring the scenario.
+        Once we implement some kind of tolerance, to allow fluctuations
+        in the rate, we can update this tests to trigger the exception
+        in a more realistic manner.
+        """
+        # XXX Update this test when we add tolerance for rate fluctuations.
+        c = Clock()
+        cluster = self.make_cluster(self.get_dropping_flocker_client_instance())
+        target_rate = 10
+        sample_size = 20
+        s = WriteRequestLoadScenario(c, cluster, request_rate=target_rate,
+                                     sample_size=sample_size)
+        dropped_rate = target_rate / 2
+        seconds_to_overload = s.max_outstanding / dropped_rate
+
+        s.start()
+        # Reach initial rate
+        cluster.get_control_service(c).drop_requests = True
+        # Initially, we generate enough dropped requests so that the scenario
+        # is overloaded when we start monitoring.
+        c.pump(repeat(1, seconds_to_overload+1))
+        # We stop dropping requests
+        cluster.get_control_service(c).drop_requests = False
+        # Now we generate the initial rate to start monitoring the scenario
+        c.pump(repeat(1, sample_size))
+        # We only need to advance one more second (first loop in the monitoring
+        # loop) to trigger RequestOverload
+        c.advance(1)
+
+        failure = self.failureResultOf(s.maintained())
+        self.assertIsInstance(failure.value, WRequestOverload)
+
+    def test_scenario_stops_only_when_no_outstanding_requests(self):
+        """
+        `ReadRequestLoadScenario` should only be considered as stopped
+        when all outstanding requests made by it have completed.
+        """
+        c = Clock()
+
+        cluster = self.make_cluster(self.get_error_response_client(c))
+        delay = 1
+
+        cluster.get_control_service(c).delay = delay
```

Comment:
Set a variable `control_service = self.get_error_response_client(c)`, and then use `control_service.delay = delay` here.

The reason is that `get_control_service()` returns a `IFlockerAPIV1Client` interface, and even though Python doesn't enforce it, we should try to live by it.